coalib/bearlib/__init__.py

```diff
@@ -82,34 +97,52 @@ def _deprecate_decorator(func):
         logged_deprecated_args = set()
 
         def wrapping_function(*args, **kwargs):
-            for arg, depr_arg_and_modifier in depr_args.items():
-                deprecated_arg, _func = (
-                    depr_arg_and_modifier
-                    if isinstance(depr_arg_and_modifier, tuple)
-                    else (depr_arg_and_modifier, lambda x: x))
+            for arg, depr_value in wrapping_function.__metadata__.depr_values:
+                if isinstance(depr_value, dict):
```

Comment:
I think this logic can also be removed into the creation of ` __metadata__` , probably by adding a new optional argument to `add_deprecated_param`